interface/modules/custom_modules/oe-module-claimrev-connect/ModuleManagerListener.php

```diff
@@ -0,0 +1,232 @@
+        }
+    }
+
+    /**
+     * Required method to return namespace
+     * If namespace isn't provided return empty string
+     * and register namespace at top of this script..
+     *
+     * @return string
+     */
+    public static function getModuleNamespace(): string
+    {
+        // Module Manager will register this namespace.
+        return 'OpenEMR\\Modules\\ClaimRevConnector\\';
+    }
+
+    /**
+     * Required method to return this class object
+     * so it will be instantiated in Laminas Manager.
+     *
+     * @return ModuleManagerListener
+     */
+    public static function initListenerSelf(): ModuleManagerListener
+    {
+        return new self();
+    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+//    private function install($modId, $currentActionStatus): mixed
+//    {
+//        $modService = new ModuleServices();
+//        /* setting the active ui flag here will allow the config button to show
+//         * before enable. This is a good thing because it allows the user to
+//         * configure the module before enabling it. However, if the module is disabled
+//         * this flag is reset by MM.
+//        */
+//        $modService::setModuleState($modId, '0', '1');
+//        return $currentActionStatus;
+//    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+    private function help_requested($modId, $currentActionStatus): mixed
+    {
+        // must call a script that implements a dialog to show help.
+        // I can't find a way to override the Lamina's UI except using a dialog.
+        if (file_exists(__DIR__ . '/show_help.php')) {
+            include __DIR__ . '/show_help.php';
+        }
+        return $currentActionStatus;
+    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+//    private function preenable($modId, $currentActionStatus): mixed
+//    {
+//        return $currentActionStatus;
+//    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+//    private function enable($modId, $currentActionStatus): mixed
+//    {
+//        $modService = new ModuleServices();
+//        if ($modService->isClaimRevConfigured()) {
+//            $modService::setModuleState($modId, '1', '0');
+//            return $currentActionStatus;
+//        }
+//        $modService::setModuleState($modId, '1', '1');
+//        return xlt("Weno eRx Service is not configured. Please configure Weno eRx Service in the Weno Module Setup.");
+//    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+//    private function disable($modId, $currentActionStatus): mixed
+//    {
+//        // allow config button to show before enable.
+//        ModuleServices::setModuleState($modId, '0', '1');
+//        return $currentActionStatus;
+//    }
```

Comment:
uncomment method and add action to disable tasks because module is now disabled. You can make this cleaner or use below. Just use same for enable action by setting active to 1. Then unregister action will delete tasks so default  table.sql will reinstall.
```
UPDATE `background_services` SET `active` = '0' WHERE `background_services`.`name` = 'ClaimRev_Elig_Send_Receive'; 
UPDATE `background_services` SET `active` = '0' WHERE `background_services`.`name` = 'ClaimRev_Receive';
UPDATE `background_services` SET `active` = '0' WHERE `background_services`.`name` = 'ClaimRev_Send';
```

Reply:
@juggernautsei You still need the enable and disable tasks i.e active flag. If user disables then background tasks don't have access to module so need to make tasks inactive. Same with reverse and enable action. You're only half way there.
See the sql example I gave above for disable.